docs/content/docs/dev/table/sql/queries/hints.md

```diff
@@ -84,4 +84,244 @@ insert into kafka_table1 /*+ OPTIONS('sink.partitioner'='round-robin') */ select
+
+### Syntax
+The `Query Hints` syntax in Flink follows the syntax of `Query Hints` in Apache Calcite:
+```sql
+# Query Hints:
+SELECT /*+ hint [, hint ] */ ...
+
+hint:
+        hintName
+    |   hintName '(' optionKey '=' optionVal [, optionKey '=' optionVal ]* ')'
+    |   hintName '(' hintOption [, hintOption ]* ')'
+
+optionKey:
+        simpleIdentifier
+    |   stringLiteral
+
+optionVal:
+        stringLiteral
+
+hintOption:
+        simpleIdentifier
+    |   numericLiteral
+    |   stringLiteral
+```
+
+### Join Hints
+
+`Join Hints` allow users to suggest the join strategy to optimizer in order to get a more optimal execution plan. 
+Now Flink `Join Hints` support `BROADCAST`, `SHUFFLE_HASH`, `SHUFFLE_MERGE` and `NEST_LOOP` Join Hints. Also, for some cases,
+`Join Hints` will conflict, we try to formulate the behavior of `Join Hints` for these cases.
+
+*NOTE:* For all types of `Join Hints`, Flink support specifying multiple tables in a single `Join Hint` or multiple `Join Hints` like
+`BROADCAST(t1, t2, ..., tn)` or `BROADCAST(t1), BROADCAST(t2), ..., BROADCAST(tn)`. During query optimization, planner will
+automatically select the optimal join strategies according to the specified `Join Hints`.
+
+#### BROADCAST
+
+{{< label Batch >}}
+
+For this type of `Join Hints`, suggests that join use `BroadCast join` strategy. It performs well when the data volume of
+one side of table is very small. The join side with the hint will be broadcast regardless of `table.optimizer.join.broadcast-threshold`.
+
+*NOTE:* `BROADCAST Join Hint` only supports join with equivalence join condition. And it doesn't support `Full Outer Join`.
+
+##### Examples
+
+```sql
+CREATE TABLE t1 (id BIGINT, name STRING, age INT) WITH (...);
+CREATE TABLE t2 (id BIGINT, name STRING, age INT) WITH (...);
+CREATE TABLE t3 (id BIGINT, name STRING, age INT) WITH (...);
+
+-- Select t1 as join side with this hint, t1 will be the broadcast table.
+SELECT /*+ BROADCAST(t1) */ * FROM t1 JOIN t2 ON t1.id = t2.id;
+
+-- Select t1 and t3 as join side with this hint.
+-- For this case, when join between t1 and t2, t1 will be the broadcast table.
+-- And when join between the result after t1 joins t2 and t3, t3 will be the broadcast table.
+SELECT /*+ BROADCAST(t1, t3) */ * FROM t1 JOIN t2 ON t1.id = t2.id JOIN t3 ON t1.id = t3.id;
+
+-- BROADCAST Join Hint don't support non-equivalent join conditions.
+-- For this case, default join strategy will be adopted instead of BroadCast join.
+SELECT /*+ BROADCAST(t1) */ * FROM t1 join t2 ON t1.id > t2.id;
+
+-- BROADCAST Join Hint strategy don't support full outer join.
+-- For this case, default join strategy will be adopted instead of BroadCast join.
+SELECT /*+ BROADCAST(t1) */ * FROM t1 FULL OUTER JOIN t2 ON t1.id = t2.id;
+```
+
+#### SHUFFLE_HASH
+
+{{< label Batch >}}
+
+For this type of `Join Hint`, suggests that join use `Shuffle Hash join` strategy. Compare of `BROADCAST`, This type of
```

Comment:
Suggests that Flink uses shuffle hash join. The join side with the hint will be the join build side, it performs well when the data volume of the hint side of table is not too large.

Reply:
> Suggests that Flink uses shuffle hash join. The join side with the hint will be the join build side, it performs well when the data volume of the hint side of table is not too large.

Done!